docs/docs/reference/other-new-features/generalized-method-syntax.md

```diff
@@ -0,0 +1,98 @@
+---
+layout: doc-page
+title: "Generalized Method Syntax"
+movedTo: https://docs.scala-lang.org/scala3/reference/other-new-features/generalized-method-syntax.html
+---
+
+The inclusion of using clauses is not the only way in which methods have been updated, type parameter clauses are now allowed in any number and at any position.
+
+## Syntax Changes
+
+### In Scala 2
+
+The old syntax only allowed zero or one type parameter clause, followed by any number of term clauses, optionnally followed by an implicit clause:
+
+```scala
+def foo[T, U](x: T)(y: U)(z: Int, s: String)(a: Array[T])(implicit List[U])
+```
+
+### In Scala 3
+
+The new syntax allows any number of type, term and using clause, in any order, optionnaly followed by an implicit clause:
+(do note however that [implicit clause are discouraged, in favor of using clauses](https://docs.scala-lang.org/scala3/reference/contextual/relationship-implicits.html))
+
+```scala
+def foo[T, U](x: T)(y: U)[V](z: V, s: String)[A](a: Array[A])(implicit List[U])
+```
+
+### Unchanged
+
+Class definitions and type declarations are unaffected, there can only be up to one type clause, in leading posion.
+
+## Motivation
+
+The new syntax is a powerful but natural extension of the old one, it allows new design patterns while staying intuitive and legible.
+
+### Dependent Type Clauses
+
+As type clauses can come after term clauses, it is now possible to have type parameters depend on term parameters:
+
+```scala
+trait Key { type Value }
+trait DB {
+  def get(k: Key): Option[k.Value] // dependent return type
+  def getOrElse(k: Key)[V >: k.Value](default: V): V // dependent type parameter
+}
+```
+
+Note that simply replacing `V` by `k.Value` would not be equivalent. For example, if `k.Value` is `Int`, the above allows: 
+`getOrElse(k)[Number](0.1)`, which returns a `Number`.
+
+### Partial Inference
+
+It is now possible to only infer some of the type parameters, this reduces boilerplate at the use site:
+```scala
+trait StaticSizeList[S <: Int & Singleton, T]
+def filled[S <: Int & Singleton][T](x: T): StaticSizeList[S,T] = ???
+val helloes = filled[4]("Hello!") // S=4, and T is inferred
+```
+
+## Details
+
+### Application
+
+Method application is unchanged, in cases where multiple type clauses are expected but not all are passed, the leftmost ones are inferred.
```

Comment:
It seems in the following example that the `C` type parameter is the one being inferred, which makes it from the right

```suggestion
Method application is unchanged, in cases where multiple type clauses are expected but not all are passed, the rightmost ones are inferred.
```

Reply:
Thanks, I added your fix as a commit, when I'll have multiple of those updates/fixes, I'll `rebase -i` to make it clean